ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -190,3 +190,43 @@ def test_indices(
         dtype=dtype[0],
         sparse=sparse,
     )
+
+    
+    @handle_frontend_test(
+        fn_tree="numpy.fill_diagonal",
+        dtype_x_indices_axis=helpers.array_indices_axis(
+            array_dtypes=helpers.get_dtypes("numeric"),
+            indices_dtypes=["int32", "int64"],
+            min_num_dims=1,
+            max_num_dims=5,
+            min_dim_size=1,
+            max_dim_size=10,
+            indices_same_dims=True,
+        ),
+        dimensions=helpers.get_shape(),
+        dtype = helpers.get_dtypes("valid", full=False),
+        sparse=helpers.get_dtypes(kind="bool"),
+        val=helpers.ints(min_value=1, max_value=10),
+        ndim=helpers.ints(min_value=2, max_value=10),
+        test_with_out=st.just(False),
+    )
+    def test_fill_diagonal(
+            *,
+            dimensions,
```

Comment:
I think you forgot to add `dtype_x_indices_axis` which you are generating in the decorator above

Reply:
Yes I wanted to use dtype_and_x